src/algebra/star/self_adjoint.lean

```diff
@@ -146,9 +155,66 @@ instance : distrib_mul_action R (self_adjoint A) :=
   smul_zero := λ r, by { ext, simp only [smul_zero', coe_smul, add_subgroup.coe_zero] } }
 
 instance : module R (self_adjoint A) :=
-{ add_smul := λ r s x, by { ext, simp only [add_smul, add_subgroup.coe_add, coe_smul] },
-  zero_smul := λ x, by { ext, simp only [coe_smul, zero_smul, add_subgroup.coe_zero] } }
+function.injective.module R (add_subgroup.subtype (self_adjoint A))
+(by { rw add_subgroup.coe_subtype, exact subtype.coe_injective })
```

Comment:
```suggestion
subtype.coe_injective
```
Should be enough here

Reply:
I went ahead and pushed these and further improvements to the `has_scalar` instances to both this PR and #12021. If we merge #12021 first, then this PR will go back to just looking like it copied `self_adjoint` to `skew_adjoint`.